tests/acceptance/new-test.js

```diff
@@ -101,6 +101,104 @@ describe('Acceptance: ember new', function () {
+    let err = await expect(ember(['new', 'foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang=.js'])).to.be
+      .rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.include('set the app programming language');
+    expect(err.message).to.include('.js');
+    expect(err.message).to.not.include('ember-cli command option');
+  });
+
+  // Misuse: possibly an attempt to set app programming language
+  it('ember new with --lang flag and edge case input: both programming language and valid language code; fails with an error message about programming language', async function () {
+    let err = await expect(ember(['new', 'foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang=css'])).to.be
+      .rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.include('set the app programming language');
+    expect(err.message).to.include('css');
+    expect(err.message).to.not.include('ember-cli command option');
+  });
+
+  // Misuse: No specification + declared option
+  it('ember new with --lang flag but no specification fails with an error message; absorbs ember-cli option (declared)', async function () {
+    let err = await expect(ember(['new', 'foo', '--skip-npm', '--skip-bower', '--lang', '--skip-git'])).to.be.rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.be.ok;
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.not.include('set the app programming language');
+    expect(err.message).to.include('ember-cli command option');
+    expect(err.message).to.include('--skip-git');
+  });
+
+  // Misuse: No specification + hidden option
+  it('ember new with --lang flag but no specification fails with an error message; absorbs appended ember-cli option (hidden)', async function () {
+    let err = await expect(ember(['new', 'foo', '--skip-npm', '--skip-bower', '--skip-git', '--lang'])).to.be.rejected;
+    expect(err.name).to.equal('SilentError');
+    expect(err.message).to.be.ok;
+    expect(err.message).to.include('An error with the `--lang` flag returned the following message:');
+    expect(err.message).to.include('Information about using the `--lang` flag:');
+    expect(err.message).to.not.include('set the app programming language');
+    expect(err.message).to.include('ember-cli command option');
+    expect(err.message).to.include('--disable-analytics');
+  });
+
```

Comment:
See above. I think we should fix this issue holistically not per-option, as such I would prefer to avoid adding specific tests for this. 

I'm happy to help someone work through a fix for the general parsing issue though...

```suggestion
```

Reply:
Removed in d3b0f8c 👍